packages/cli/commands/upload.js

```diff
@@ -204,6 +209,33 @@ exports.handler = async options => {
       absoluteSrcPath,
       options.convertFields
     );
+
+    if (options.remove) {
+      //  If remove is true, will first delete the dest folder and then upload src. Cleans up files that only exist on HS.
+      const remoteFileList = await getDirectoryContentsByPath(accountId, dest);
```

Comment:
Hmm I'm a little iffy on this prompting strategy, leads to kinda weird scenario like here - I'm prompted for if I want to delete `package.json` specifically, and if I say no, it's still overwritten, so my answer kinda didn't matter...

<img width="582" alt="Screenshot 2023-07-28 at 2 26 07 PM" src="https://github.com/HubSpot/hubspot-cli/assets/25852903/f8fb63a5-8547-4d04-802d-eab77b0452c1">

I think we might be OK, instead of getting directory contents from the dest and prompting about deleting each, to just prompt about deleting the contents at `dest` once for the whole directory and blasting the whole thing in one go. 

FWIW, the `--remove` on `hs watch` doesn't prompt at all. So I don't think we need to be _too_ in depth with this

Reply:
Yeah, I was thinking of using "overwrite" for naming as opposed to remove, but it's also kind of awkward.

I'm fine with just prompting once before the command is run. Probably can just remove `--force` in that case.

Reply:
cc @markhazlewood  -- we were discussing this internally but wanted to get your opinion on the language used here for the option `--overwrite` as opposed to other suggestions that you might have for the functionality. We don't have a designer on our side and since you were heading up the CLI guidelines originally, I thought you'd be a good confirmation here.

Reply:
@TanyaScales Thanks for looping me in! Not to throw a wrench in, but when I see "removes the original and completely overwrites it with this one" my mind immediately goes to `--clean`. A "clean" operation (clean build, or clean install, etc.) usually means to completely remove the current one and replace it, rather than trying to merge them together.

Reply:
Hmm yeah overwrite is a little awkward too. When I hear it my assumption is that it overwrites the remote files with my versions, not deletes ones that aren't there. I think we should keep it as `--remove` to remain in line with `hs watch`'s option since it's effectively the same behavior. Some of the copy for that can maybe be borrowed.

Reply:
Actually, the `--remove` on `hs watch` functions slightly differently doesn't it! My assumption without checking code was that it removes on the initial upload _AND_ any files you delete while its running but it actually only does the second. Should we change that?

Reply:
(If I'm reading this right) For the `--remove` flag on watch -- if you also flag `--initial-upload` are you saying that files that don't exist locally aren't removed on that initial upload to the DM?

Also, the suggestion here is to have `hs watch --remove` and `hs upload --remove` to behave the same -- With the obvious difference being the underlying function of the main command? And that flag behavior being -- removal of files remotely that don't exist locally.

Additionally, using the `--initial-upload` flag paired with the `--remove` flag on the `watch` command would then do, basically, an "overwrite/clean" of the dest (mirroring `hs upload --remove`).

I think I am for making the removal of files behave the same across the board in the two commands. Purely because I think that it keep consistency across the cli in what that particular word means for file management.

Reply:
>For the --remove flag on watch -- if you also flag --initial-upload are you saying that files that don't exist locally aren't removed on that initial upload to the DM?

Yep, never noticed it before while working on CAT but I figured it from reading the code - just ran it on growth theme to check: 
<img width="844" alt="Screenshot 2023-08-14 at 11 50 04 AM" src="https://github.com/HubSpot/hubspot-cli/assets/25852903/71974c1f-87ad-4873-8354-989e765a2c93">
<img width="372" alt="Screenshot 2023-08-14 at 11 58 08 AM" src="https://github.com/HubSpot/hubspot-cli/assets/25852903/30b2bfd8-1b45-4b8c-a8a0-30a8562c72ab">

>Also, the suggestion here is to have hs watch --remove and hs upload --remove to behave the same -- With the obvious difference being the underlying function of the main command? And that flag behavior being -- removal of files remotely that don't exist locally.
>Additionally, using the --initial-upload flag paired with the --remove flag on the watch command would then do, basically, an "overwrite/clean" of the dest (mirroring hs upload --remove).
>I think I am for making the removal of files behave the same across the board in the two commands. Purely because I think that it keep consistency across the cli in what that particular word means for file management.

Yeah this is my feeling. If the two are going to have the same argument name they should either both do a "full clean" of the remote directory, or if they're going to be different then have different names. 

Is it possible we'd be breaking something by changing this? I wonder if there are devs that rely on it behaving like this & keep certain files in their DFS but not locally and just expect them to stay there. Maybe pretty niche use case.

Also, should it only do the initial removal if you use both `--initial-upload` AND `--remove`? Or is just the `--remove` enough? Is there a better way to distinguish these behavior options than combinations of these two args?

Reply:
> Is it possible we'd be breaking something by changing this? I wonder if there are devs that rely on it behaving like this & keep certain files in their DFS but not locally and just expect them to stay there. Maybe pretty niche use case.

This would be my concern about changing it -- however, we can be pretty brazen about communicating the change in a version update, I think. Those developers should probably not be using the `--remove` flag in that case altogether. (IMO)...

I agree with pairing down to one argument that does _one_ thing -- not two different scenarios.

> Also, should it only do the initial removal if you use both --initial-upload AND --remove? Or is just the --remove enough? Is there a better way to distinguish these behavior options than combinations of these two args?

Is there a better way to distinguish these? That is a good question. I do feel like there isn't a great combinator of two, maybe `hs watch --clean` like Mark was suggesting. That would mean it could upload all files and remove the ones that aren't present locally. But, that leaves the door open to having `hs watch --initial-upload` still exist for those folks that don't want to use the removal feature. And, to me, the two args `--initial-upload` and `--remove` are a little more self-explanatory about what is going to happen?

Reply:
Also, on another note, a question: Should we be moving forward with this as it is and determine some changes that make sense to the CLI in subsequent issues? I feel like there are quite a few things that we could determine need refactoring or more thought put into if we really wanted to. The current solution does solve the issue at hand -- even if it is a bit disjointed. Without having to go into multiple refactors during this PR -- maybe we should consider a different issue to bring the two removal options more in line with each other?